Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/calculation/DccWalletInfoCalculation.kt

```diff
@@ -1,17 +1,98 @@
+    @BaseGson private val gson: Gson
+) {
+
+    private lateinit var jsonFunctions: JsonFunctions
+    private var boosterRulesNode: JsonNode = NullNode.instance
+
+    fun init(
         cclConfiguration: CCLConfiguration,
-        dccList: List<DccData<*>>
+        boosterRules: List<DccValidationRule>
+    ) {
+        jsonFunctions = JsonFunctions()
+        boosterRulesNode = gson.toJson(boosterRules).toJsonNode()
+        cclConfiguration.logic.jfnDescriptors.forEach {
+            jsonFunctions.registerFunction(it.name, it.definition.toJsonNode())
+        }
+    }
+
+    fun getDccWalletInfo(
+        dccList: List<CwaCovidCertificate>
     ): DccWalletInfo {
-        val defaultInputParameters = getDefaultInputParameters()
-        return dummyDccWalletInfo
+
+        val input = getDccWalletInfoInput(dccList = dccList).toJsonNode()
+        val output = jsonFunctions.evaluateFunction(
+            FUNCTION_NAME,
+            input
+        )
+
+        return mapper.treeToValue(output, DccWalletInfo::class.java)
+    }
+
+    @VisibleForTesting
+    internal fun getDccWalletInfoInput(
+        dccList: List<CwaCovidCertificate>,
+        boosterNotificationRules: JsonNode = boosterRulesNode,
+        defaultInputParameters: CclInputParameters = getDefaultInputParameters(),
+    ) = DccWalletInfoInput(
+        os = defaultInputParameters.os,
+        language = defaultInputParameters.language,
+        now = SystemTime(
+            timestamp = defaultInputParameters.now.timestamp,
+            localDate = defaultInputParameters.now.localDate,
+            localDateTime = defaultInputParameters.now.localDateTime,
+            localDateTimeMidnight = defaultInputParameters.now.localDateTimeMidnight,
+            utcDate = defaultInputParameters.now.utcDate,
+            utcDateTime = defaultInputParameters.now.utcDateTime,
+            utcDateTimeMidnight = defaultInputParameters.now.utcDateTimeMidnight,
+        ),
+        certificates = dccList.toCclCertificateList(),
+        boosterNotificationRules = boosterNotificationRules
+    )
+
+    private fun List<CwaCovidCertificate>.toCclCertificateList(): List<CclCertificate> {
+        return filter {
+            it.getState() != CwaCovidCertificate.State.Recycled
+        }.map {
+            CclCertificate(
+                barcodeData = it.qrCodeToDisplay.content,
+                cose = Cose(it.dccData.kid),
+                cwt = Cwt(
+                    iss = it.headerIssuer,
+                    iat = it.headerIssuedAt.millis / 1000,
+                    exp = it.headerExpiresAt.millis / 1000
+                ),
+                hcert = it.dccData.certificateJson.toJsonNode(),
+                validityState = it.getState().toCclState()
+            )
+        }
     }
+
+    private fun Any.toJsonNode(): JsonNode = mapper.valueToTree(this)
+
+    private fun String.toJsonNode(): JsonNode = mapper.readTree(this)
 }
+
+private const val FUNCTION_NAME = "getDccWalletInfo"
```

Comment:
Tech spec defines it as `getDCCWalletInfo` DCC instead of Dcc. Can this be a problem?

Reply:
Tech spec is wrong

Reply:
name in the config is getDccWalletInfo

Reply:
I see :)